iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -770,7 +746,8 @@ class expected<ValueType, ErrorType>
     ///         })
     /// @endcode
     ///
-    template <typename Optional = ValueType, typename std::enable_if<is_optional<Optional>::value, int>::type = 0>
+    template <typename OptionalType = ValueType,
+              typename std::enable_if<is_optional<OptionalType>::value, int>::type = 0>
     const expected& if_empty(const cxx::function_ref<void()>& callable) const noexcept;
```

Comment:
I would like to get rid of this method and replace it with something more generic instead of having it tied to something considered "optional".
i.e. identify when the type has an `and_then()` method (without arguments) and if it does, allow an `and_then()` to be specified by the expected